syn/syn.ys

```diff
@@ -0,0 +1,18 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# yosys synthesis script
+
+read -sv prim_clock_gating.v syn_out/*.v
+synth -top ibex_core
```

Comment:
Do we want -flatten here?  I've prefered a flattened hierarchy for probing around the design.  When you leave  the modules it this can be confusing (I'm not sure tracing logic cones and such goes past module boundaries or I remember having trouble with this, the longest topological path command, ltp, also doesn't seem to deal with it well declaring you have loops in your design because there's some cell that takes input from a module as well as outputs to it).

I suspect flattening gives better synthesis results also.

Reply:
Sure, I can add -flatten. I'm just using the example synthesis script given in the Yosys documentation.

Alternatively, do you want to LGTM this PR and then you can create a follow-on PR with all your improvements?